purchases/src/main/kotlin/com/revenuecat/purchases/PostReceiptHelper.kt

```diff
@@ -84,15 +96,24 @@ internal class PostReceiptHelper(
             receiptInfo = receiptInfo,
             storeUserID = purchase.storeUserID,
             marketplace = purchase.marketplace,
-            onSuccess = { info, _ ->
+            onSuccess = { info ->
                 billing.consumeAndSave(finishTransactions, purchase)
                 onSuccess?.let { it(purchase, info) }
             },
-            onError = { error, shouldConsumePurchase, _, _ ->
+            onError = { error, shouldConsumePurchase, isServerError, _ ->
                 if (shouldConsumePurchase) {
                     billing.consumeAndSave(finishTransactions, purchase)
                 }
-                onError?.let { it(purchase, error) }
+                useOfflineEntitlementsCustomerInfoIfNeeded(
+                    isServerError,
+                    appUserID,
+                    onSuccess = { customerInfo ->
+                        onSuccess?.let { it(purchase, customerInfo) }
+                    },
+                    onError = {
+                        onError?.let { it(purchase, error) }
+                    }
+                )
```

Comment:
This is pretty similar between both `post` methods here. Originally I tried to move it to the common `postReceiptAndSubscriberAttributes` method in this class, but that was tricky since in that case, we would be calling the `onSuccess` blocks in these callers which always consume/save the sent token in cache... We don't want to consume purchases on offline entitlements mode, so I had to separate this so it still went through the `onError` blocks. It's possible this can be refactored better, but I think this is ok for now.